src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -624,6 +624,28 @@ public UserDetails loadUserByUsername(String username)
     }
 
     /**
+     * Compare an object against this instance for equivalence.
+     * @param object An object to campare this instance to.
+     * @return true if the objects are the same instance and configuration.
+     */
+    @Override
+    public boolean equals(Object object){
+        if(object instanceof GithubSecurityRealm) {
+            GithubSecurityRealm obj = (GithubSecurityRealm) object;
```

Comment:
`GitHubSecurityRealm` isn't final, so these could in fact still be different. I doubt that's relevant in practice though.